tests/components/group/test_notify.py

```diff
@@ -90,3 +93,58 @@ def test_send_message_with_data(self):
             "title": "Test notification",
             "data": {"hello": "world", "test": "message"},
         }
+
+
+async def test_reload_notify(hass):
+    """Verify we can reload the notify service."""
+
+    assert await async_setup_component(
+        hass,
+        "group",
+        {},
+    )
+    await hass.async_block_till_done()
+
+    assert await async_setup_component(
+        hass,
+        notify.DOMAIN,
+        {
+            notify.DOMAIN: [
+                {"name": "demo1", "platform": "demo"},
+                {"name": "demo2", "platform": "demo"},
+                {
+                    "name": "group_notify",
+                    "platform": "group",
+                    "services": [{"service": "demo1"}],
+                },
+            ]
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert hass.services.has_service(notify.DOMAIN, "demo1")
+    assert hass.services.has_service(notify.DOMAIN, "demo2")
+    assert hass.services.has_service(notify.DOMAIN, "group_notify")
+
+    yaml_path = path.join(
+        _get_fixtures_base_path(),
+        "fixtures",
+        "group/configuration.yaml",
+    )
+    with patch.object(hass_config, "YAML_CONFIG_FILE", yaml_path):
+        await hass.services.async_call(
+            "group",
+            SERVICE_RELOAD,
+            {},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+
+    assert hass.services.has_service(notify.DOMAIN, "demo1")
+    assert hass.services.has_service(notify.DOMAIN, "demo2")
+    assert not hass.services.has_service(notify.DOMAIN, "group_notify")
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6103615)